File: sample/src/main/java/com/kandoka/springframework/beans/factory/config/SingletonBeanRegistry.java
package com.kandoka.springframework.beans.factory.config;

/**
 * @Description 单例缓存 接口
 * @Author handong3
 * @Date 2022/2/15 16:20
 */
public interface SingletonBeanRegistry {

    /**
     * 获取单例Bean
     * @param beanName bean名称
     * @return
     */
    Object getSingleton(String beanName);

    void registerSingleton(String beanName, Object singletonObject);
}

File: core/src/main/java/com/kandoka/core/beans/factory/config/SingletonBeanRegistry.kt
package com.kandoka.core.beans.factory.config

/**
 *
 *  @Description 定义了一个Bean的单例注册中心
 *  @Author handong3
 *  @Date 2022/11/11 9:36
 *
 */
interface SingletonBeanRegistry {
    fun getSingleton(beanName: String): Any;
    fun registerSingleton(beanName: String, singletonObject: Any);
}